src/v6_indexer/utils.ts

```diff
@@ -480,4 +501,225 @@ export async function getVaultBalances(
+  // // // Pool 
+  // // // oracle: TwapOracle
+  // // // // TwapOracle 
+  // // // // aggregator 
+  // // // // last_updated_timestamp
+  // // // // created_at_timestamp
+  // // // // last_price 
+  // // // // last_observation
+  // // // // max_observation_change_per_update
+  // // // // initial_observtion 
+  // // // // start_delay_seconds
+  // // // quote_reserves 
+  // // // base_reserves
+  // // // quote_protocol_fee_balance
+  // // // base_protocol_fee_balance
+  // total_liquidity: u128
+  // base_mint
+  // quote_mint
+  // amm_base_vault
+  // amm_quote_vault
+
+  // LaunchProposalEvent gives:
+  // proposal key 
+  // dao key
+  // total_staked 
+  // post_amm_state which is dao.amm.clone() feels like a lot can come from this.
+
+  // ProvideLiquidityEvent gives:
+  // dao key
+  // liquidity provider
+  // position_authority
+  // quote_amount 
+  // base_amount 
+  // liquidity_minted
+  // min_liqudity
+  // post_amm_state which is dao.amm.clone(), gives good stuff
+  
+  // WithdrawLiquidityEvent 
+  // dao key 
+  // liqudity_provider
+  // liqudity_withdrawn 
+  // min_base_amount 
+  // min_quote_amount 
+  // base_amount
+  // quote_amount 
+  // post_amm_state
+
+  // ConditionalSwapEvent
+  // dao key 
+  // proposal key
+  // trader key
+  // market 
+  // swap_type 
+  // input_amount 
+  // output_amount 
+  // min_output_amount 
+  // post_amm_state
+
+  // SpotSwapEvent 
+  // dao key 
+  // user key 
+  // swap_type 
+  // input_amount 
+  // output_amount 
+  // min_output_amount 
+  // post_amm_state 
+
+  // common is on all and gives 
+  // Slot 
+  // unix_timestamp
+  // dao_seq_num
+
+  logger.info("insertPriceIfNotDuplicate::event", event);
+  const existingPrice = await db.select()
+    .from(schema.prices)
+    .where(and(
+      eq(schema.prices.marketAcct, event.dao.toBase58()),
+      eq(schema.prices.updatedSlot, event.common.slot.toString())
+    ))
+    .limit(4);
+
+  if (existingPrice.length > 3) {
+    logger.info("Price already exists", event.dao.toBase58(), BigInt(event.common.slot.toString()));
+    return;
+  }
+
+  // Ensure tokens exist in database 
+  await Promise.all([
+    insertTokenIfNotExists(db, event.postAmmState.baseMint),
+    insertTokenIfNotExists(db, event.postAmmState.quoteMint)
+  ]);
+
+  // Extract pools from the futarchy AMM state
+  const futarchyState = event.postAmmState.state.futarchy;
+  
+  if (!futarchyState) {
+    logger.warn("No futarchy state found in AMM");
+    return;
+  }
+
+  const passPool = futarchyState.pass;
+  const failPool = futarchyState.fail;
+
+  // Get token decimals from the AMM state
+  const baseMint = event.postAmmState.baseMint.toString();
+  const quoteMint = event.postAmmState.quoteMint.toString();
+  
+  const [baseDecimals, quoteDecimals] = await Promise.all([
+    getTokenDecimals(db, baseMint),
+    getTokenDecimals(db, quoteMint)
+  ]);
+
+  const calculatePrice = (pool: unknown): BN => {
```

Comment:
Consider the current indexer ability and see if we have to architect something new to handle this given the SpotSwap would technically pass through these conditional markets (I think..)